addons/website_slides/controllers/main.py

```diff
@@ -434,8 +433,9 @@ def channel(self, channel, category=None, tag=None, page=1, slide_type=None, unc
         # documentation mode may display less slides than content by category but overhead of
         # computation is reasonable
         values['slide_promoted'] = request.env['slide.slide'].sudo().search(domain, limit=1, order=order)
+        category_slides_domain = domain + [('id', '!=', values.get('slide_promoted').id)] if values.get('slide_promoted') else domain
```

Comment:
It seems too much complicated. Why not showing a promoted slide twice (once in promoted, once in its category) ? I would remove all constraints and simplify layout and display.